website/source/docs/agent/http/operator.html.markdown

```diff
@@ -130,3 +134,140 @@ If ACLs are enabled, the client will need to supply an ACL Token with
 
 The return code will indicate success or failure.
 
+### <a name="keyring-install"></a> /v1/operator/keyring/install
+
+Available in Consul 0.7.2 and later, the keyring install endpoint supports the
+`PUT` method.
+
+#### PUT Method
+
+Using the `PUT` method, this endpoint will install a new gossip encryption key
+into the cluster. There is more information on gossip encryption available
+[here](/docs/agent/encryption.html#gossip-encryption).
+
+The register endpoint expects a JSON request body to be PUT. The request
+body must look like:
+
+```javascript
+{
+  "Key": "3lg9DxVfKNzI8O+IQ5Ek+Q=="
+}
+```
+
+The `Key` field is mandatory and provides the encryption key to install into the
+cluster.
+
+If ACLs are enabled, the client will need to supply an ACL Token with
+[`keyring`](/docs/internals/acl.html#keyring) write privileges.
+
+The return code will indicate success or failure.
+
+### <a name="keyring-list"></a> /v1/operator/keyring/list
+
+Available in Consul 0.7.2 and later, the keyring install endpoint supports the
+`GET` method.
+
+#### GET Method
+
+Using the `GET` method, this endpoint will list the gossip encryption keys
+installed on both the WAN and LAN rings of every known datacenter. There is more
+information on gossip encryption available
+[here](/docs/agent/encryption.html#gossip-encryption).
+
+If ACLs are enabled, the client will need to supply an ACL Token with
+[`keyring`](/docs/internals/acl.html#keyring) read privileges.
+
+A JSON body is returned that looks like this:
+
+```javascript
```

Comment:
It might be nice to add an endpoint here (or in the API client based on this) that summarizes the keys that have been successfully installed everywhere and are safe to switch to. Probably not in this pass, but it would be nice to know as an operator in a very high level way "this key is good to go".